waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBossJob.js

```diff
@@ -68,17 +68,29 @@ class PgBossJob extends Job {
  * functions, we will override the previous calls.
  * @param {string} jobName - The user-defined job name in their .wasp file.
  * @param {fn} jobFn - The user-defined async job callback function.
- * @param {object} defaultJobOptions - PgBoss specific options for boss.send() applied to every submit() invocation,
+ * @param {object} defaultJobOptions - pg-boss specific options for boss.send() applied to every submit() invocation,
  *                                     which can overriden in that call.
+ * @param {object} jobSchedule [Optional] - The cron string and arguments/options when invoking the job.
  */
-export async function createJob({ jobName, jobFn, defaultJobOptions } = {}) {
-  // As a safety precaution against undefined behavior of registering different
-  // functions for the same job name, remove all registered functions first.
-  await boss.offWork(jobName)
+export async function createJob({ jobName, jobFn, defaultJobOptions, jobSchedule } = {}) {
+  // Note: createJob runs before pg-boss starts. Therefore, anything that expects pg-boss to be running
+  // should be registered as a callback passed to registerAfterStartCallback.
+  registerAfterStartCallback(async () => {
+    // As a safety precaution against undefined behavior of registering different
+    // functions for the same job name, remove all registered functions first.
+    await boss.offWork(jobName)
 
-  // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
-  // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
-  await boss.work(jobName, jobFn)
+    // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
+    await boss.work(jobName, jobFn)
+
+    // If a job schedule is provided, we should schedule the recurring job.
+    // If the schedule name already exists, it's updated to the newly provided cron expression, arguments, and options.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#scheduling
+    if (jobSchedule !== null) {
```

Comment:
The template generation code ensures it is either `null` or an `object`, as `jobSchedule` is optional. We have quirks in the Mustache template process where we cannot pass a full object, so I had to explicitly either render it into the template or set it to `null`. It should never be `undefined`.

We can do the `if (jobSchedule) {`, although I tend to dislike implicit false checks but I think it is the more JS way so all good. :) Per your cron note, even though it should be there if the template passes the `jobSchedule`, we can do something like ``if (jobSchedule && jobSchedule.cron) {` even for extra safety, just in case the Haskell changes/breaks that assumption in the future.

Reply:
Ah got it -> yeah I tend to look at code ignoring any guarantees made by the generator, so in this case what I just see is that if somebody passes undefined for that arg, stuff will break. I get the dislike of implicit false checks but I think they can actually be pretty convenient when used carefully! I think `if (jobSchedule) {` would be enough here, since so that code calling this function is ok if they don't specify `jobSchedule` - it doesn't crash.

Reply:
BTW, some people like to use `jobSchedule != null` (notice the lack of a third `=` sign). It performs type coercion and returns `false` for anything except `null` and `undefined`. I've never been a fan of it though, just putting it out there.